src/Servers/Kestrel/Transport.Sockets/src/Internal/SocketConnection.cs

```diff
@@ -209,7 +216,18 @@ private async Task ProcessReceives()
 
                 input.Advance(bytesReceived);
 
-                var flushTask = input.FlushAsync();
+                var flushTask = input.FlushAsync(isMoreData: !_waitForData);
```

Comment:
@benaadams @davidfowl Any chance we're getting the `isMoreData:` API in .NET 5?

Reply:
No; too soon